debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/JdbcChangeEventSink.java

```diff
@@ -129,8 +129,7 @@ public void execute(Collection<SinkRecord> records) {
                     flushBufferWithRetries(collectionId, upsertBufferByTable.get(collectionId).flush());
                 }
 
-                Buffer tableIdBuffer = resolveBuffer(deleteBufferByTable, collectionId, record);
-                List<JdbcSinkRecord> toFlush = tableIdBuffer.add(record);
+                List<JdbcSinkRecord> toFlush = getRecordsToFlush(upsertBufferByTable, collectionId, record);
```

Comment:
```suggestion
                List<JdbcSinkRecord> toFlush = getRecordsToFlush(deleteBufferByTable, collectionId, record);
```
I believe there was a copy-n-paste error, no?